src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -0,0 +1,687 @@
+      std::make_pair(donor, entry) : std::make_pair(entry, donor);
+    auto [liter, riter] = is_left ?
+      std::make_pair(donor_iter, iter) : std::make_pair(iter, donor_iter);
+    if (donor->extent_is_below_min()) {
+      logger().debug("{}::merge_entry make_full_merge l {} r {}", __func__, *l, *r);
+      assert(entry->extent_is_below_min());
+      return l->make_full_merge(oc, r).safe_then([=] (auto &&replacement){
+        journal_inner_update(liter, replacement->get_laddr(), maybe_get_delta_buffer());
+        journal_inner_remove(riter, maybe_get_delta_buffer());
+        //retire extent
+        std::list<laddr_t> dec_laddrs;
+        dec_laddrs.push_back(l->get_laddr());
+        dec_laddrs.push_back(r->get_laddr());
+        return oc.tm.dec_ref(oc.t, dec_laddrs).safe_then([this, oc] (auto &&ret) {
+          if (extent_is_below_min() &&
+              get_node_meta().depth != oc.omap_root.depth) { //not root
+            return merge_entry_ret(
+                   merge_entry_ertr::ready_future_marker{},
+                   mutation_result_t(mutation_status_t::NEED_MERGE, std::nullopt,
+                                    this->cast<OMapNode>()));
+          } else {
+            return merge_entry_ret(
+                   merge_entry_ertr::ready_future_marker{},
+                   mutation_result_t(mutation_status_t::SUCCESS, std::nullopt, std::nullopt));
+          }
+        });
+      });
+    } else {
+      logger().debug("{}::merge_entry balanced l {} r {}", __func__, *l, *r);
+      return l->make_balanced(oc, r, !is_left).safe_then([=] (auto tuple) {
+        auto [replacement_l, replacement_r, replacement_pivot] = tuple;
+        if (!extent_will_overflow(replacement_pivot.size() + 1, std::nullopt)) {
+          journal_inner_update(liter, replacement_l->get_laddr(), maybe_get_delta_buffer());
+          journal_inner_replace(riter, replacement_r->get_laddr(),
+                                replacement_pivot, maybe_get_delta_buffer());
+          std::list<laddr_t> dec_laddrs;
+          dec_laddrs.push_back(l->get_laddr());
+          dec_laddrs.push_back(r->get_laddr());
+          return oc.tm.dec_ref(oc.t, dec_laddrs).safe_then([] (auto &&ret) {
+            return merge_entry_ret(
+                   merge_entry_ertr::ready_future_marker{},
+                   mutation_result_t(mutation_status_t::SUCCESS, std::nullopt, std::nullopt));
+          });
+        } else {
+          logger().debug("{}::merge_entry balanced and split {} r {}", __func__, *l, *r);
+          return make_split_children(oc).safe_then([=] (auto p_tuple) {
+            OMapNodeRef pleft = std::get<0>(p_tuple);
+            OMapNodeRef pright = std::get<1>(p_tuple);
+            std::string p_pivot = std::get<2>(p_tuple);
+            OMapInnerNodeRef p_left = pleft->cast<OMapInnerNode>();
+            OMapInnerNodeRef p_right = pright->cast<OMapInnerNode>();
+            auto [replacement_l, replacement_r, replacement_pivot] = tuple;
```

Comment:
This is pretty much identical to the code in insert.  Please factor these out as a common insert_inner helper.

Reply:
Perhaps something like this?

```
/**
 * inner_replace
 *
 * Replaces entries in [begin, end) with the contents of replacements.  May result in a split or merge outcome
 * encoded in the returned mutation_result_t
 */
ertr::future<mutation_result_t>
inner_replace(iterator begin, iterator end, std::vector<std::pair<std::string, laddr_t>> replacements);
```

Reply:
rename to make_replace similar as make_balanced make_entry etc.